### PR TITLE
4.6

### PR DESCRIPTION
**Pull request by koneill94**
All playbooks pulled from RH deployment repo except manifests.yaml which deals with all install-config, manifest and ignition config generation. Doesn't include the running of the deployment playbooks.

### CODE REVIEW

**Comment by koneill94** [comment]
Has been updated to include server groups and a deploy script

File: bootstrap.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import base64
```
Comment (andrew-garner) [inline bootstrap.py:1]: A short comment at the top of this file would be useful, just to outline what it does. Maybe a python3 shebang too.

File: master.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+for index in $(seq 0 2); do
```
Comment (andrew-garner) [inline master.sh:1]: I don't see why this is a shell script? Couldn't the loop be done in python?

File: deploy.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+ansible-playbook -e @vars.yml generate-inventory.yaml
```
Comment (andrew-garner) [inline deploy.sh:1]: `set -e` would be useful here too I think

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# file showing needed dependancies for runs
+server-groups.yaml needs 'ansible-galaxy collection install openstack.cloud' ansible 2.9 or greater
```
Comment (andrew-garner) [inline requirements.txt:2]: This file could be a yaml file with this in.

```yaml
---
collections:
- name: openstack.cloud
  version: "1.2.0"
```

See https://docs.ansible.com/ansible/latest/galaxy/user_guide.html#installing-roles-and-collections-from-the-same-requirements-yml-file

File: vars.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+apiFIP: <FIP for controlplane>
+ingressFIP: <FIP for dataplane>
+bootstrapFIP: <FIP for bootstrap access>
+custID: <Estate API ID for cluster>
+baseDomain: <Base domain for OSP region>
+rhcosImage: <rhcos image present in OSP project>
+externalDNS: <List of 2 DNS servers>
+apiAllowedSources: "0.0.0.0/0"
+ingressAllowedSources: "0.0.0.0/0"
+sshAllowedSources: "0.0.0.0/0"
```
Comment (andrew-garner) [inline vars.yml:12]: Do we have any safer defaults we could use here?

File: pre-run-files.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+- hosts: localhost
```
Comment (andrew-garner) [inline pre-run-files.yaml:1]: This could be called generate-clouds.yaml to match the generate-inventory playbook.

File: security-groups.yaml [hunk 7]
```diff
@@ -0,0 +1,417 @@
...
+  - name: 'Set SSH security group tag'
+    command:
+      cmd: "openstack security group set --tag {{ cluster_id_tag }} {{ os_sg_ssh }} "
+
+  - name: 'Create master-sg rule "ICMP"'
+    os_security_group_rule:
+      security_group: "{{ os_sg_master }}"
+      protocol: icmp
+
+  - name: 'Create master-sg rule "machine config server"'
```
Comment (andrew-garner) [inline security-groups.yaml:66]: Some of these could use with_items which would make the file shorter and the list easier to maintain. If this is working now though I'm not suggesting we change it all now.

**Review by andrew-garner — COMMENTED** [review]
Comments added inline.

File: master.sh (hunk 2 above)
Comment (koneill94) [reply master.sh:1]: This is just copied from the install instructions - don't want to change what they've given us if it works

File: requirements.txt (hunk 4 above)
Comment (koneill94) [reply requirements.txt:2]: Nice, this has been added along with a requirements install line to the deploy.sh

File: vars.yml (hunk 5 above)
Comment (koneill94) [reply vars.yml:12]: The control plane and dataplane should be open to all sources by default but I've restricted the ssh sources to office and VPN networks only

**Comment by koneill94** [comment]
Changed to address comments. Sec groups we pass in can now be lists too.

**Review by andrew-garner — APPROVED** [review]
I'm not an expert on whether all those security groups are correct, but as far as I can see all looks good. Thanks for the changes about my comments. LGTM

**Pull request merged**